app/src/main/java/com/example/xpvehicles/fragments/FilterDialogFragment.java

```diff
@@ -61,6 +95,67 @@ public void onClick(View v) {
                 dismiss();
             }
         });
+    }
+
+    private void setShowVehiclesOnClickListener() {
+        btnFilterShowVehicles.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String maxPrice = String.valueOf(etMaxPrice.getText());
+                String minPrice = String.valueOf(etMinPrice.getText());
+                String maxDistance = String.valueOf(etMaxDistance.getText());
+                queryFilterVehicles(maxDistance, minPrice, maxPrice);
+            }
+        });
+    }
+
+    private void queryFilterVehicles(String maxDistance, String minPrice, String maxPrice) {
+        final String QUERY_PARAMETER_OWNER = "owner";
+
+        ParseQuery<Vehicle> query = ParseQuery.getQuery(Vehicle.class);
+        query.whereNotEqualTo(QUERY_PARAMETER_OWNER, ParseUser.getCurrentUser().getObjectId());
+
+        // add query for max price
+        if (!maxPrice.isEmpty()) {
+            query.whereLessThanOrEqualTo("dailyPrice", parseDouble(maxPrice));
+        }
+        // add query for min price
+        if (!minPrice.isEmpty()) {
+            query.whereGreaterThanOrEqualTo("dailyPrice", parseDouble(minPrice));
+        }
+        // add query for max distance
+        if (!maxDistance.isEmpty()){
+            query.whereWithinMiles("geoLocation", activity.getUserLocationGeoPoint(), parseDouble(maxDistance));
+        }
+
+        query.findInBackground(new FindCallback<Vehicle>() {
+            @Override
+            public void done(List<Vehicle> vehicles, ParseException e) {
+                if (e != null) {
+                    Log.e(TAG, "Issue with filtering vehicles",e);
+                    return;
+                }
+                exploreAdapter.clear();
+                if (vehicles.size() > 0) {
+                    tvNoAvailableRentVehicle.setVisibility(View.GONE);
+                    exploreAdapter.addAll(vehicles);
```

Comment:
Why don't we just call exploreAdapter.setVehicles()? Let the adapter to handle if it is empty and needs to clear